Rust/Rust.sublime-syntax

```diff
@@ -896,17 +1144,17 @@ contexts:
       pop: true
 
   fn-where:
-    - meta_scope: meta.function.rust
+    - meta_scope: meta.function.rust meta.where.rust
     - include: comments
     - match: '(?=\{)'
       set: fn-body
-    - match: \bfor\b
-      scope: keyword.other.rust
     - match: \bwhere\b
       scope: keyword.other.rust
     - include: type-any-identifier
-    - match: ':'
+    - match: '[:,]'
       scope: punctuation.separator.rust
+    - match: ';'
```

Comment:
you could use a lookahead here to pop, or correctly set the scope to `punctuation.terminator.rust`.

Reply:
Thanks! I pushed a fix to use lookahead, and fixed another problem where it was not handling `+` in trait bounds correctly.